### ISSUE TITLE
Function parameters misinterpreted as fragment interpolation

### ISSUE THREAD

**Issue by andreaswachowski** [apollographql/eslint-plugin-graphql#155]
Consider the following case (using eslint-plugin-graphql 2.1.1):

`blob.js`:

```js
import gql from 'graphql-tag';

export const getBlob = (blobId, thumbnailSize) => gql`
  {
    blob(id: "${blobId}") {
      id,
      thumbnail(size: "${thumbnailSize}") {
        url
      }
    }
  }
`;
```

`.eslintrc.js` contains:

```js
'graphql/template-strings': [
  'error',
  {
    env: 'apollo',
    schemaJson: require('./schema.json')
  }
]
```

This produces
```
5:17  error  Invalid interpolation - fragment interpolation must occur outside of the brackets  graphql/template-strings
```

(5:17 is the `b` from `blobId` in the line `blob(id: "${blobId}") {`.

I've just stumbled upon this and there might be other, better ways to implement this and workaround the linter error. Nevertheless, it looks to me that an error should not be reported.

**Comment by lrojas94** [comment]
going through the same :+1:

**Comment by jnwng** [comment]
in this case, this is guarding against dynamic interpolation in the graphql template literal itself — the expected way to parameterize a graphql query looks [like this](https://graphql.org/learn/queries/#variables)
```graphql
query ($blobId: ID!, $thumbnailSize) {
  blob(id: $blobId) {
    id,
    thumbnail(size: $thumbnailSize) {
      url
    }
  }
}
```
which also makes sure you're using the right types and constructing a valid GraphQL AST. this is general guidance that we think protects folks from errant interpolation, especially in the `apollo` env.

do you think a different environment should be included that allows for arbitrary interpolation?

(closing for now, this is intended behavior for the `apollo` environment)

**Comment by onsah** [comment]
Related to this: If I have a constant defined somewhere as I am using it in the application and I want to hardcode it into the query as well, how I can achieve it? Do I still have to define it as query parameter and pass everywhere I am performing the query? Ideally I could just interpolate it into query string so I don't need to think about it when I am querying.
